### PR TITLE
fix: don't ask to download a snapshot if one is already given

### PR DESCRIPTION
**Pull request by lemmih**
**Summary of changes**
Changes introduced in this pull request:
- Don't ask to download a snapshot if we're already trying to import one.

**Reference issue to close (if applicable)**

Closes 

**Other information and links**

### CODE REVIEW

File: forest/daemon/src/daemon.rs [hunk 1]
```diff
@@ -184,20 +184,6 @@ pub(super) async fn start(config: Config, detached: bool) {
     .unwrap();
     chain_store.set_genesis(&genesis.blocks()[0]).unwrap();
 
-    // Terminate if no snapshot is provided or DB isn't recent enough
```
Comment (hanabi1224) [inline forest/daemon/src/daemon.rs:187]: Maybe also re-enable the `download` and `import` part of `calibnet sync check` job?

Comment (lemmih) [reply forest/daemon/src/daemon.rs:187]: I'd rather merge support for NV17 and then re-enable the entire sync check.

**Review by creativcoder — APPROVED** [review]

**Review by hanabi1224 — APPROVED** [review]

**Comment by LesnyRumcajs** [comment]
This is not exactly right, the prompt should be given before everything else. Otherwise there's a race condition in the stdout.

image

**Comment by LesnyRumcajs** [comment]
Instead, perhaps the condition in the `prompt_snapshot_or_die` should be adapted to this case I missed, so omit download if `import_snapshot` is already set to something.

**Review by LesnyRumcajs — APPROVED** [review]
Looks good. Thanks for fixing this.

**Pull request merged**